### PR TITLE
MINOR: update Kafka Streams standby task config

### PR DESCRIPTION
**Pull request by mjsax**
We see many real-time deployments that benefit highly from standby tasks. We should bump it's priority to `HIGH` and add as recommended configuration.

Call for review @ableegoldman @astubbs

### CODE REVIEW

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -755,7 +759,7 @@ <h4><a class="toc-backref" href="#id30">probing.rebalance.interval.ms</a><a clas
               <dl class="docutils">
                 <dt>Recommendation:</dt>
                 <dd>Increase the replication factor to 3 to ensure that the internal Kafka Streams topic can tolerate up to 2 broker failures.
-                  Note that you will require more storage space as well (3 times more with the replication factor of 3).</dd>
+                  Note that you will require more storage space as well (3x with the replication factor of 3).</dd>
```
Comment (ableegoldman) [inline docs/streams/developer-guide/config-streams.html:835]: Why this change?

**Review by ableegoldman — APPROVED** [review]
LGTM, thanks @mjsax !

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -1025,12 +1034,19 @@ <h4><a class="toc-backref" href="#id22">acks</a><a class="headerlink" href="#ack
...
-          <pre class="line-numbers"><code class="language-java">Properties streamsSettings = new Properties();
-streamsSettings.put(StreamsConfig.REPLICATION_FACTOR_CONFIG, 3);
+        </div>
+        <div class="section" id="i32">
+          <h4><a class="toc-backref" href="#id23">num.standby.replicas</a><a class="headerlink" href="#id2" title="Permalink to this headline"></a></h4>
+          <blockquote>
+            <div>See the <a class="reference internal" href="#streams-developer-guide-standby-replicas"><span class="std std-ref">description here</span></a>.</div></blockquote>
+        </div>
+        <pre class="line-numbers"><code class="language-java">Properties streamsSettings = new Properties();
+// for broker version is 2.3. or older
```
Comment (showuon) [inline docs/streams/developer-guide/config-streams.html]: additional dot is added.
// for broker version is 2.3[.] or older

File: docs/streams/developer-guide/config-streams.html [hunk 3]
```diff
@@ -680,14 +680,18 @@ <h4><a class="toc-backref" href="#id33">default.windowed.value.serde.inner</a><a
               preferred to restart on an instance that has standby replicas so that the local state store restoration process from its
               changelog can be minimized.  Details about how Kafka Streams makes use of the standby replicas to minimize the cost of
               resuming tasks on failover can be found in the <a class="reference internal" href="../architecture.html#streams_architecture_state"><span class="std std-ref">State</span></a> section.
+              <dl class="docutils">
+                <dt>Recommendation:</dt>
+                <dd>Increase the number of standbys to 1 to get instant fail-over, ie, high-availability.
```
Comment (showuon) [inline docs/streams/developer-guide/config-streams.html]: ie, high-availability. -> i.e., high-availability.

**Review by showuon — COMMENTED** [review]
Thanks for the PR. Left some comments. Thanks.

File: docs/streams/developer-guide/config-streams.html (hunk 3 above)
Comment (astubbs) [inline docs/streams/developer-guide/config-streams.html]: @showuon did you know you can directly make suggestions? much faster and more convenient than writing it out how you did.
```suggestion
                <dd>Increase the number of standbys to 1 to get instant fail-over, i.e., high-availability.
```

**Comment by astubbs** [comment]
Only other suggestion would be to actually reference / mention the section for resiliency at the top of the page? So that people know it exists. People don't typically read an entire page of config options - so they normally won't "happen across" the CRTICIAL section at the bottom for production deployments.

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html:835]: `3 times more` sounds incorrect, it's only 2x _more_ OR 3 times...

And we use `3x` somewhere else.

**Comment by mjsax** [comment]
@astubbs There is a TOC at the top of the page... Not sure what else we could do?

Screen Shot 2021-10-20 at 4 42 03 PM

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html:835]: Ah, fair enough. Was just wondering since we use the phrasing `twice as much with 1 standby` above -- maybe there we should update to say `2x as much with 1 standby` for consistency?

Super trivial though, so feel free to ignore

**Comment by astubbs** [comment]
> @astubbs There is a TOC at the top of the page... Not sure what else we could do?

Yeah, but for such _critical_ settings, it's way too subtle. People don't read carefully entire TOCs. Two options (not that I'm a doc designer or anything) - perhaps 
a) but a `WARNING:: please note read section blah blah` at the top - like an [asciidoc admonitions](https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#admonitions)
b) actually move the section to be the first section, as it's far more important to read than _anything_ else on that page - for example, if you forget to put an app id your app will crash immediately, bu the resiliency settings are not obvious until you have yourself a sev1 in prod.

File: docs/streams/developer-guide/config-streams.html [hunk 4]
```diff
@@ -680,14 +680,18 @@ <h4><a class="toc-backref" href="#id33">default.windowed.value.serde.inner</a><a
               preferred to restart on an instance that has standby replicas so that the local state store restoration process from its
               changelog can be minimized.  Details about how Kafka Streams makes use of the standby replicas to minimize the cost of
               resuming tasks on failover can be found in the <a class="reference internal" href="../architecture.html#streams_architecture_state"><span class="std std-ref">State</span></a> section.
+              <dl class="docutils">
+                <dt>Recommendation:</dt>
+                <dd>Increase the number of standbys to 1 to get instant fail-over, i.e., high-availability.
+                  Note that you will require more client side storage space as well (twice as much with 1 standby).</dd>
```
Comment (JimGalasyn) [inline docs/streams/developer-guide/config-streams.html]:
```suggestion
                  Increasing the number of standbys requires more client-side storage space. For example, with 1 standby, twice as much space is required.</dd>
```

**Comment by JimGalasyn** [comment]
> > @astubbs There is a TOC at the top of the page... Not sure what else we could do?
> 
> b) actually move the section to be the first section, as it's far more important to read than _anything_ else on that page

We could move this section under Required Settings and note that although not required, it's extremely important (or critical) to set `num.standby.replicas` to 1 to ensure high availability.

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few suggestions.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `3.1` branch.
